lib/Doctrine/ORM/Query/AST/WhereClause.php

```diff
@@ -13,15 +13,9 @@
  */
 class WhereClause extends Node
 {
-    /** @var ConditionalExpression|ConditionalTerm */
-    public $conditionalExpression;
-
-    /**
-     * @param ConditionalExpression $conditionalExpression
-     */
-    public function __construct($conditionalExpression)
+    /** @param ConditionalExpression|ConditionalTerm $conditionalExpression */
```

Comment:
Allowing A|B in the property but just A with the constructor causes many issues, I don't think it's worth the hassle to keep it.

Reply:
At least not in this PR. 882 files are quite tiring to read. :sweat_smile:. New type hint PRs will follow for sure for 3.0 in the future.